client/components/theme-tier/constants.js

```diff
@@ -4,11 +4,34 @@ import {
 	PLAN_PERSONAL,
 	PLAN_PREMIUM,
 } from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
 
-export const THEME_TIER_TO_PLAN = {
-	free: PLAN_FREE,
-	personal: PLAN_PERSONAL,
-	premium: PLAN_PREMIUM,
-	partner: PLAN_BUSINESS,
-	woocommerce: PLAN_BUSINESS,
+// This should be kept consistent with the theme_tier taxonomy.
+export const THEME_TIERS = {
+	free: {
+		label: translate( 'Free' ),
+		plan: PLAN_FREE,
+	},
+	personal: {
+		label: translate( 'Personal' ),
+		plan: PLAN_PERSONAL,
+	},
+	premium: {
+		label: translate( 'Premium' ),
+		plan: PLAN_PREMIUM,
+	},
+	partner: {
+		label: translate( 'Partner', {
+			context: 'This theme is developed and supported by a theme partner',
+		} ),
+		plan: PLAN_BUSINESS,
+	},
+	woocommerce: {
+		label: translate( 'WooCommerce' ),
+		plan: PLAN_BUSINESS,
+	},
+	sensei: {
+		label: translate( 'Sensei' ),
+		plan: PLAN_BUSINESS,
+	},
```

Comment:
Open question (cc @Automattic/t-rex):

We need a hardcoded list of tier slugs in Calypso to handle the `:tier` routing needs of the Showcase.

But, do we also need to define the localized labels and associated plans here?
Should we do it server-side?

(This shouldn't be a blocker for this PR, but something that we might want to iterate upon.)

Reply:
Localization should occur at the uppermost layer, in this case, Calypso. The API should be agnostic regarding the front-end translation needs. There might be different FEs with different needs, besides, we would need the API to know what locale to translate things to which could lead to increased caching rules and complexity.

Ideally, since we're working with two different systems, we should request the tiers DTO response from the API, then map the tiers DTO to the internal model/DTO we want to use in Calypso, and render. That way both the FE and BE can evolve without being tightly coupled to each other and leaking the API domain into Calypso.

Reply:
> Localization should occur at the uppermost layer, in this case, Calypso. The API should be agnostic regarding the front-end translation needs. There might be different FEs with different needs, besides, we would need the API to know what locale to translate things to which could lead to increased caching rules and complexity.

Some endpoints are locale-aware. I think that all theme endpoints are localized, since they fetch data (theme descriptions, taxonomies) localized on the Showcase site.

Either way, I agree with you that we should translate at the last possible moment whenever possible — like in this case.

> Ideally, since we're working with two different systems, we should request the tiers DTO response from the API, then map the tiers DTO to the internal model/DTO we want to use in Calypso, and render. That way both the FE and BE can evolve without being tightly coupled to each other and leaking the API domain into Calypso.

Yep, although the way the whole theme system is built doesn't make this exactly straightforward. 😄 

I'd say that mapping to plans — specifically, mapping to **plan bundles** (e.g. the "personal" bundle includes "free, personal monthly, personal yearly, personal bi-yearly") — is something that only really makes sense on the front end for labelling and filtering reasons.

All the checks and guards are performed on features, not plans.

Reply:
> All the checks and guards are performed on features, not plans.
This to me is a smell that maybe we haven't modeled tiers correctly.

Reply:
> This to me is a smell that maybe we haven't modeled tiers correctly. But as long as we can correctly map features to plans on the FE it should be fine, it will increase maintenance costs, but it should be manageable.

@rcrdortiz got it!
I've accidentally introduced confusion with the `THEME_TIERS.plan` property.
It's not for gating but for upselling.

It means: "the lowest plan needed to activate the theme".

Ideally, and assuming that plans are always strictly hierarchical, we'd have some kind of `getMinimumPlanForFeature` in the `calypso-products` package.

But for now, I guess it's enough to clarify that the `THEME_TIERS.plan` property is for UI purposes.
WDYT renaming it as `minimumUpsellPlan` (plus an inline comment)? It's a mouthful, but it should encourage folks to investigate further, instead of just using it without thinking twice.

Reply:
> WDYT renaming it as minimumUpsellPlan (plus an inline comment)? 

I like it, but what do you think about `nextUpsellPlan`? I think it's a bit more concise and has pretty much the same meaning. Either way, both are an improvement so feel free to use any of them.

Reply:
Just pushed, so my version will stick (at least until I go on holiday and can't complain if you change it 😄 ).